runtime-extension-model/src/main/java/org/mule/runtime/core/api/extension/MuleExtensionModelDeclarer.java

```diff
@@ -1394,4 +1408,9 @@ private void declareSecurityFilter(ExtensionDeclarer extensionDeclarer) {
         .describedAs("The name of the encryption strategy to use. This should be configured using the "
             + "'password-encryption-strategy' element, inside a 'security-manager' element at the top level.");
   }
+
+  public static void refreshSystemProperties() {
+    SUPPORT_EXPRESSIONS_IN_VARIABLE_NAME_IN_SET_VARIABLE =
+        parseBoolean(getProperty(SUPPORT_EXPRESSIONS_IN_VARIABLE_NAME_IN_SET_VARIABLE_PROPERTY, "true"));
```

Comment:
use getBoolean with default false.  and rename the variable to indicate that this is a kill switch. REVERT_SUPPORT_EXPRESSIONS_N_SET_VARIABLE_PROPERTY.
Boolean.getBoolean(REVERT_SUPPORT_EXPRESSIONS_N_SET_VARIABLE_PROPERTY)

Reply:
As far as I know getBoolean does not seem to support adding a default value, so I need to keep `parseBoolean(getProperty...`